File: app/src/main/java/com/mytlogos/enterprise/viewmodel/NotificationViewModel.java
package com.mytlogos.enterprise.viewmodel;

import android.app.Application;

import androidx.lifecycle.LiveData;
import androidx.paging.PagedList;

import com.mytlogos.enterprise.background.TaskManager;
import com.mytlogos.enterprise.model.NotificationItem;

public class NotificationViewModel extends RepoViewModel {

    NotificationViewModel(Application application) {
        super(application);
    }

    public LiveData<PagedList<NotificationItem>> getNotifications() {
        return this.repository.getNotifications();
    }

    public void clearNotifications() {
        TaskManager.runTask(this.repository::clearNotifications);
    }
}

File: app/src/main/java/com/mytlogos/enterprise/viewmodel/NotificationViewModel.kt
package com.mytlogos.enterprise.viewmodel

import android.app.Application
import androidx.lifecycle.viewModelScope
import androidx.paging.cachedIn
import com.mytlogos.enterprise.background.repository.NotificationRepository

class NotificationViewModel(application: Application) : RepoViewModel(application) {
    val notificationRepository by lazy { NotificationRepository.getInstance(application) }
    val notifications by lazy { notificationRepository.notifications.cachedIn(viewModelScope) }

    suspend fun clearNotifications() {
        notificationRepository.clearNotifications()
    }
}